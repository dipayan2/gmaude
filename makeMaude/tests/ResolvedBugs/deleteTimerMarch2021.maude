set show timing off .
set show advisories off .

***
***	Test for bug where deleteTimer() left the interpreter in an
***	inconsistent state and caused a crash on the next rewrite
***	command.
***

load time

mod TEST is
  inc TIME .
  op myClass : -> Cid .
  ops me : -> Oid .

vars O O2 O3 : Oid .
  rl < O : myClass | none > createdTimer(O, O2, O3) =>
     < O : myClass | none > deleteTimer(O3, me) .
endm

erew <> < me : myClass | none > createTimer(timeManager, me) .
red 0 .
