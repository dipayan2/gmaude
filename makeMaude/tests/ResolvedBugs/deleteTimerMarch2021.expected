==========================================
erewrite in TEST : <> createTimer(timeManager, me) < me : myClass | none > .
rewrites: 1
result Configuration: <> deletedTimer(me, timer(0)) < me : myClass | none >
==========================================
reduce in TEST : 0 .
rewrites: 0
result Zero: 0
Bye.
